src/wallpapers/linux.rs

```diff
@@ -0,0 +1,177 @@
+                Command::new("xfconf-query")
+                    .args(&[
+                        "-c",
+                        "xfce4-desktop",
+                        "-p",
+                        "/backdrop/screen0/monitor0/workspace0/last-image",
+                        "-s",
+                        &xfce_path,
+                    ])
+                    .output()?;
+            }
+
+            DesktopEnvt::Deepin => {
+                Command::new("dconf")
+                    .args(&[
+                        "write",
+                        "/com/deepin/wrap/gnome/desktop/background/picture-uri",
+                        &path,
+                    ])
+                    .output()?;
+            }
+
+            DesktopEnvt::Kde => {
+                // KDE needs plasma shell scripting to change the wallpaper
+                let kde_set_arg = format!(
+                    r#"
+                    const monitors = desktops()
+                    for (var i = 0; i < monitors.length; i++) {{
+                        monitors[i].currentConfigGroup = ["Wallpaper"]
+                        monitors[i].writeConfig("Image", {})
+                    }}"#,
+                    &path
+                );
+
+                Command::new("qdbus")
+                    .args(&[
+                        "org.kde.plasmashell",
+                        "/PlasmaShell",
+                        "org.kde.PlasmaShell.evaluateScript",
+                        &kde_set_arg,
+                    ])
+                    .output()?;
+            }
+        }
+
+        Ok(())
+    }
+
+    fn get_wallpaper(&self) -> Result<PathBuf, Box<dyn Error>> {
+        let output = match self {
+            DesktopEnvt::Gnome => Command::new("gsettings")
+                .args(&["get", "org.gnome.desktop.background", "picture-uri"])
+                .output()?,
+
+            DesktopEnvt::Cinnamon => Command::new("dconf")
+                .arg("read")
+                .arg("/org/cinnamon/desktop/background/picture-uri")
+                .output()?,
+
+            DesktopEnvt::Mate => Command::new("dconf")
+                .args(&["read", "/org/mate/desktop/background/picture-filename"])
+                .output()?,
+
+            DesktopEnvt::Xfce => Command::new("xfconf-query")
+                .args(&[
+                    "-c",
+                    "xfce4-desktop",
+                    "-p",
+                    "/backdrop/screen0/monitor0/workspace0/last-image",
+                ])
+                .output()?,
+
+            DesktopEnvt::Deepin => Command::new("dconf")
+                .args(&[
+                    "read",
+                    "/com/deepin/wrap/gnome/desktop/background/picture-uri",
+                ])
+                .output()?,
+
+            DesktopEnvt::Kde => return Ok(kde_get()?),
+        };
+
+        let output = enquote::unquote(String::from_utf8(output.stdout)?.trim().into())?;
+        Ok(PathBuf::from(output))
+    }
+}
+
```

Comment:
Comment on the purpose on these functions

Reply:
I added documentation to the trait.